### ISSUE TITLE
Fix coercion of Strings/Lists/Dict in Append

### ISSUE THREAD

**Issue by ajf58** [ajf58/tigris-sandbox#2962]
This issue was originally created at: 2014-07-13 05:32:13.
This issue was reported by: `dirkbaechle`.
dirkbaechle said at 2014-07-13 05:32:13
>As reported on the mailing list:
> 
>  http://scons.tigris.org/ds/viewMessage.do?
>dsForumId=1272&viewType=browseAll&dsMessageId=2876001
> 
>the env.Append() method can yield surprising results, when mixing data types 
>like Strings and Lists for the arguments.
> 
>This issue should add support for the most basic use cases:
> 
>1.)
> 
>env.Append(CCFLAGS = '-O1')
>env.Append(CCFLAGS = '-whatever')
> 
>(expected is CCFLAGS = ['-O1','-whatever']
> 
>2.)
> 
>env = Environment(CCFLAGS='breakit')
>env.Append(CCFLAGS = ['-g3'])
>env.Append(CCFLAGS = 'test')
> 
>3.)
> 
>env = Environment(CCFLAGS=['breakit'])
>env.Append(CCFLAGS = ['-g3'])
>env.Append(CCFLAGS = 'test')
> 
>, and also includes proper documentation about how coercions are handled in 
>general (coerce upward from String to List|Dict, no mix of List+Dict supported).

dirkbaechle said at 2014-07-13 05:32:44
>added dependency

dirkbaechle said at 2014-07-13 07:33:46
>*** Issue 2599 has been marked as a duplicate of this issue. ***

dirkbaechle said at 2014-07-13 07:35:24
>added dependency

dirkbaechle said this issue depends on #2900 at 2014-07-13 05:32:44.

dirkbaechle said this issue depends on #2926 at 2014-07-13 07:35:24.

dirkbaechle said this issue blocks #2900 at 2014-07-13 05:32:44.

dirkbaechle said this issue blocks #2926 at 2014-07-13 07:35:24.

dirkbaechle said this issue  is a duplicate of #2900 at 2014-07-13 05:32:44.

dirkbaechle said this issue  is a duplicate of #2926 at 2014-07-13 07:35:24.

dirkbaechle said this issue is duplicated by #2900 at 2014-07-13 05:32:44.

dirkbaechle said this issue is duplicated by #2926 at 2014-07-13 07:35:24.
